src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * @dev Updates yield parameters.
+     * Callable only by the contract owner / proxy admin.
+     * @param _yieldParams new yield parameters.
+     */
+    function updateYieldParams(YieldParams memory _yieldParams) external onlyOwner {
+        address yieldAddress = yieldParams.yield;
+        require(
+            _yieldParams.yield == yieldAddress || investedAssetsAmount == 0,
+            "ZkBobCompoundingPool: Invested amount should be 0 in case of changing yield"
+        );
+        require(
+            _yieldParams.yield == address(0) || _yieldParams.interestReceiver != address(0),
+            "ZkBobCompoundingPool: interest receiver should not be address(0) for existed yield"
+        );
+
+        if (_yieldParams.yield != yieldAddress) {
+            if (_yieldParams.yield != address(0)) {
+                IERC20(token).approve(_yieldParams.yield, type(uint256).max);
+            }
+            if (yieldAddress != address(0)) {
+                IERC20(token).approve(yieldAddress, 0);
+            }
+        }
+
+        yieldParams = _yieldParams;
+
+        emit UpdateYieldParams(_yieldParams);
+    }
+
+    /**
+     * @dev Rebalances yield bearing tokens.
+     * @param minRebalanceAmount minimum amount of token to move between underlying balance and yield.
+     * @param maxRebalanceAmount maximum amount of token to move between underlying balance and yield.
+     */
+    function rebalance(uint256 minRebalanceAmount, uint256 maxRebalanceAmount) external {
+        if (maxRebalanceAmount < minRebalanceAmount) {
+            (minRebalanceAmount, maxRebalanceAmount) = (maxRebalanceAmount, minRebalanceAmount);
+        }
+
+        YieldParams storage params = yieldParams;
+        IERC4626 yieldVault = IERC4626(params.yield);
+
+        if (address(yieldVault) == address(0)) {
+            return;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender,
+            "ZkBobCompoundingPool: Rebalance is an operator-called method"
+        );
+
+        uint256 underlyingBalance = IERC20(token).balanceOf(address(this));
+        uint256 buffer = params.buffer;
+
+        uint256 balancesDiff = (underlyingBalance >= buffer) ? underlyingBalance - buffer : buffer - underlyingBalance;
+        balancesDiff = (balancesDiff >= minRebalanceAmount)
+            ? ((balancesDiff <= maxRebalanceAmount) ? balancesDiff : maxRebalanceAmount)
+            : 0;
+        if (balancesDiff <= params.dust) {
+            return;
+        }
+        if (underlyingBalance > buffer) {
+            investedAssetsAmount += balancesDiff;
+            yieldVault.deposit(balancesDiff, address(this));
+            emit Rebalance(0, balancesDiff);
+        } else {
+            investedAssetsAmount -= balancesDiff;
+            yieldVault.withdraw(balancesDiff, address(this), address(this));
+            emit Rebalance(balancesDiff, 0);
+        }
```

Comment:
Seems that case when `investedAssetsAmount < balancesDiff` is missing.
```suggestion
        if (underlyingBalance > buffer + minRebalanceAmount) {
            uint256 balancesDiff = underlyingBalance - buffer;
            if (balancesDiff > maxRebalanceAmount) balancesDiff = maxRebalanceAmount; 
            investedAssetsAmount += balancesDiff;
            yieldVault.deposit(balancesDiff, address(this));
            emit Rebalance(0, balancesDiff);
        } else if (underlyingBalance + minRebalanceAmount < buffer) {
            uint256 balancesDiff = buffer - underlyingBalance - minRebalanceAmount;
            if (balancesDiff > maxRebalanceAmount) balancesDiff = maxRebalanceAmount;
            if (balancesDiff > investedAssetsAmount) balancesDiff = investedAssetsAmount;
            investedAssetsAmount -= balancesDiff;
            yieldVault.withdraw(balancesDiff, address(this), address(this));
            emit Rebalance(balancesDiff, 0);
        }
```

Reply:
Current code is complicated to understand and read due to large number of nested if branches and ternary conditions. Please, consider refactoring this piece and adding the comments to help better understand what's going on here.
I've given a couple of suggestions in the above comment.